assets/js/event-providers/mailchimp.js

```diff
@@ -14,14 +14,67 @@
  * limitations under the License.
  */
 
+import { classifyPII, getUserData } from './utils';
+
 ( ( mc4wp ) => {
 	if ( ! mc4wp ) {
 		return;
 	}
 
-	mc4wp.forms.on( 'subscribed', () => {
+	mc4wp.forms.on( 'subscribed', ( mc4wpForm, data ) => {
+		const gtagUserDataEnabled = global._googlesitekit?.gtagUserData;
+
+		const userData = gtagUserDataEnabled
+			? getUserDataFromForm( mc4wpForm.element, data )
+			: null;
+
 		global._googlesitekit?.gtagEvent?.( 'submit_lead_form', {
 			event_category: 'mailchimp',
+			...( userData ? { user_data: userData } : {} ),
 		} );
 	} );
 } )( global.mc4wp );
+
+/**
+ * Extracts and classifies user data from a Mailchimp form submission.
+ *
+ * @since n.e.x.t
+ *
+ * @param {HTMLFormElement} form The submitted form element.
+ * @param {Object}          data The submitted form's data.
+ * @return {Object|undefined} A user_data object containing detected PII (address, email, phone_number), or undefined if no PII found.
+ */
+function getUserDataFromForm( form, data ) {
+	// eslint-disable-next-line sitekit/acronym-case
+	if ( ! form || ! ( form instanceof HTMLFormElement ) ) {
+		return undefined;
+	}
+
+	const detectedFields = Object.entries( data )
+		.map( ( [ name, value ] ) => {
+			// Mailchimp joins the individual name fields into a single field "NAME",
+			// but still provides the individual values in the data object, so we skip them.
+			if ( 'NAME' in data && ( name === 'FNAME' || name === 'LNAME' ) ) {
+				return null;
+			}
```

Comment:
I understand the reasoning behind this, but I feel this is not the accurate path to take. Instead, as we already have the first name and last name explicitly defined, we should instead skip `NAME` and use the explicit values instead.

Take my name as an example, which is `Nahid Ferdous` as the first name, and `Mohit` as the last name.

However, following the current process, the user data is sent as `nahid` as `first_name`, and `ferdous mohit` as `last_name`, which is incorrect.

```suggestion
			// Mailchimp joins the individual name fields into a single field "NAME",
			// but still provides the individual values in the data object.
			// We only rely on "NAME" when neither "FNAME" nor "LNAME" are available.
			if ( name === 'NAME' && ( 'FNAME' in data || 'LNAME' in data ) ) {
				return null;
			}
```